src/easylink/step.py

```diff
@@ -14,4 +20,15 @@ class Step:
     """
 
     name: str
+    prev_input: bool  # Toy version of TODO [MIC-4774]
+    input_files: list  # Toy version of TODO [MIC-4774]
     input_validator: Callable = validate_input_file_dummy
+
+    def get_subgraph(self, config: "Config") -> MultiDiGraph:
+        sub = MultiDiGraph()
+        sub.add_node(
```

Comment:
I don't see where add_node is defined - How is the networkx.MultiDiGraph.add_node method supporting such custom things as Implementation() as args?

Reply:
`add_node` lets you define the names of attributes while you're passing them in! I didn't think about that this is kind of neat. https://networkx.org/documentation/stable/reference/classes/generated/networkx.Graph.add_node.html

So if I say
`sub.add_node(node_obj, attr1=foo, attr2=bar)`
then I'm telling it that the node as an attribute `attr1` with value `foo`. The node objects and attributes can be any python object (the nodes also have to be hashable)